### ISSUE TITLE
Variable's type cannot be updated 

### ISSUE THREAD

**Issue by zhangwenx** [PecanProject/bety#309]
Browser: Firefox 
Go to Data -> Variables -> Edit record -> select a random Type -> Update -> click on Show: The type is not updated, always be empty

**Comment by dlebauer** [comment]
I've reproduced this error. The following appears in `log/production.log`:

```
args = []; block = 
  Rendered variables/edit.html.erb within layouts/application (8.9ms)
  Rendered feedback/_fbtab.html.erb (8.3ms)
Completed 200 OK in 40.4ms (Views: 37.4ms | ActiveRecord: 0.7ms)
Started PUT "/variables/571" for 128.174.125.237 at 2015-06-03 15:31:54 -0500
Processing by VariablesController#update as HTML
  Parameters: {"utf8"=>"✓", "authenticity_token"=>"ixQy+vFhE8sonrc9D9YhNsm/eBAYI8q0w1JkcxVCvkE=", "variable"=>{"name"=>"A", "standard_name"=>"", "units"=>"umol m2 s", "standard_units"=>"", "description"=>"Assimilation Rate", "label"=>"", "type"=>"ecosystem", "notes"=>"Photosynthesis rate. Requires covariates temperature and PAR"}, "commit"=>"Update", "id"=>"571"}
Redirected to https://www.betydb.org/variables/571
Completed 302 Found in 5.8ms (ActiveRecord: 0.8ms)
Started GET "/variables/571" for 128.174.125.237 at 2015-06-03 15:31:54 -0500
Processing by VariablesController#show as HTML
  Parameters: {"id"=>"571"}
********************************************************************************
args = [{}]; block = 
  Rendered variables/show.html.erb within layouts/application (1.1ms)
  Rendered feedback/_fbtab.html.erb (8.5ms)
Completed 200 OK in 32.6ms (Views: 29.7ms | ActiveRecord: 0.6ms)
Started GET "/variables/571/edit" for 128.174.125.237 at 2015-06-03 15:31:58 -0500
Processing by VariablesController#edit as HTML
  Parameters: {"id"=>"571"}
********************************************************************************
args = []; block = 
  Rendered variables/edit.html.erb within layouts/application (8.7ms)
  Rendered feedback/_fbtab.html.erb (8.3ms)
Completed 200 OK in 57.3ms (Views: 54.4ms | ActiveRecord: 0.6ms)
```

**Comment by gsrohde** [comment]
Fixed in commit 0a6aad9ffa7 (once deployed).

**Comment by gsrohde** [comment]
Fixed in release 4.2.
